[OSPO Book] Update 05-chapter.md - OSPO Metrics with CHAOSS

germonprez opened the pull request:
This is the first PR for Chapter 5 content.

Watch the preview [here](https://github.com/todogroup/ospology/blob/cca7e569864f243138e921177e2103dae781e4cc/ospo-book/content/en/05-chapter.md)

anajsana commented:
Thank you @germonprez ! I will allocate some time during our two contributor's call agenda next Monday to let people go through the content and review together

PS: AFAIK, I remember there were some visuals included in the google doc version. would it be possible to add it to the PR to?

germonprez commented:
Yes. I can add the one visual. Where do you want images placed? Is there a central folder for chapter images?

anajsana commented:
> Yes. I can add the one visual. Where do you want images placed? Is there a central folder for chapter images?

Yes! all images should be stored here: https://github.com/todogroup/ospology/tree/main/ospo-book/static/images

P.S. I still need to clean up the folders to move the images displayed in the book to the correct folder

lpanni reviewed (approved):
Great work, well written chapter!

germonprez commented:
I have updated the PR to include the things discussed in our book meeting today. Specifically: 

1. Include links to the CHAOSS Practitioner Guides
2. Include Impact image 
3. Updated the order of the Impact narrative to align with the image in point 2 above.

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
```diff
@@ -1,7 +1,100 @@
-# TBD
+## Introduction
+
+Metrics for metrics sake benefit no one. The average age of issues is 10.3 days. The total number of pull requests was 121 last month. We had 3 new companies join our community over the past 15 days. Without context, these metrics provide no insight, yet we are driven to reach for metrics to provide insight into our complex open source engagements. 
```
```suggestion
Metrics for metrics sake benefit no one. Consider these metrics: The average age of issues is 10.3 days. The total number of pull requests was 121 last month. We had 3 new companies join our community over the past 15 days. Without context, these metrics provide no insight, yet we are driven to reach for metrics to provide insight into our complex open source engagements. 
```

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
```diff
@@ -1,7 +1,100 @@
+
+Metrics for metrics sake benefit no one. The average age of issues is 10.3 days. The total number of pull requests was 121 last month. We had 3 new companies join our community over the past 15 days. Without context, these metrics provide no insight, yet we are driven to reach for metrics to provide insight into our complex open source engagements. 
+
+There are a number of reasons why we need insights into open source projects. an organization built from open source would like to track contributions back to key projects. an organization that participates in an open source ecosystem wants to recognize potential failures and provide stabilizing resources as needed. an organization wants to do their part to ensure the longevity of open source sof ...
```
```suggestion
There are a number of reasons why we need insights into open source projects. Some example reasons are: An organization built from open source would like to track contributions back to key projects. An organization that participates in an open source ecosystem wants to recognize potential failures and provide stabilizing resources as needed. An organization wants to do their part to ensure the longevity of open source software, in particular, the software that is meaningful to their business. And, of course, an organization needs to remain compliant with upstream license requirements and attend to security issues that can impact the ways they work.
```

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
```diff
@@ -1,7 +1,100 @@
+
+## Measuring and Communicating Impact
+
+Beyond helping understand one open source project or a collection of projects, metrics can also play an important role in communicating impact. Finding consistent and meaningful ways to communicate impact is important for any organization unit, including open source program offices. Following the goal-question-metric approach used in the CHAOSS project, we present four goals that open source prog ...
```
```suggestion
Open source program offices (OSPOs), like any organizational unit, need consistent and meaningful ways to communicate impact. Metrics help us to understand one open source project or a collection of projects, and they also play an important role in communicating impact. Following the goal-question-metric approach used in the CHAOSS project, we present four goals that open source program offices can consider, alongside associated questions that can provide insight to the associated goals. Stemming from these goals and questions, we recommend a series of metric-related CHAOSS [Practitioner Guides](https://chaoss.community/about-chaoss-practitioner-guides/) to provide specific recommendations within an organization.
```

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
```diff
@@ -1,7 +1,100 @@
+
+### Communicating Impact: Open Source
+
+Beyond helping understand one open source project or a collection of projects, metrics can also play an important role in communicating impact. Finding consistent and meaningful ways to communicate impact is important for any organization unit, including open source program offices. Following approaches used in the CHAOSS project, we present four goals that open source program offices can conside ...
```
This is a duplicate of the paragraph above.

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
```diff
@@ -1,7 +1,100 @@
+
+The proposed process' focus on semi-automating the health assessments, where open source components at the intake stage are manually inspected using standardized checklists with automated tool support as needed. Inspections need to be lightweight yet rigorous enough to capture the concerned health aspects of importance. A lightweight documentation process is also used to persist and index analysi ...
+
+A quantitative screening and monitoring process is additionally required for open source components already integrated and deployed. This process should idealy be automated as large amounts of dependencies commonly exist, and manual overviews and inspections are not applicable. Tooling should ideally be customized and integrated into CI/CD pipelines or run on regular occasions.
```
```suggestion
A quantitative screening and monitoring process is additionally required for open source components already integrated and deployed. This process should ideally be automated as a large number of dependencies commonly exist, and manual overviews and inspections are not practical. Tooling should ideally be customized and integrated into CI/CD pipelines or run on regular occasions.
```

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
```diff
@@ -1,7 +1,100 @@
+
+#### Goal 4: Organizational Impact
+
+Engagement with open source communities includes working in the upstream to effectively use open source software in organizational products. In this, there is a need to monitor the intake of open source software for infosec, legal, and engineering reasons. Companies can establish software intake processes, working with teams to either technically track or socially consider issues related to open  ...
```
```suggestion
Engagement with open source communities includes working in the upstream to effectively use open source software in organizational products. In this, there is a need to monitor the intake of open source software for infosec, legal, and engineering reasons. Companies can establish software intake processes, working with teams to either technically track or socially consider issues related to open source intake. Organization impact can also include working downstream with projects and companies that rely on your organizational products. This can include working to gain a clearer picture of  the open source that is in your shipped products. Organizations can work in securing and regulating their own internal open source processes in an effort to improve product development activities. Key questions to consider regarding organization impact include: 
```

alice-sowerby commented on ospo-book/content/en/05-chapter.md:
I note that the chapter name is now different, it may be worth harmonizing it with the TOC so they align. https://ospobook.todogroup.org/toc/

anajsana commented:
@alice-sowerby is this PR ready for merge? please note that early next year (probably January or mid February) , there will be a review meeting to go through all chapters at once for final review and cross-referencing

alice-sowerby commented:
@anajsana I have put some time in my calendar for this Friday to catch up on the book stuff! Will respond more fully then.

alice-sowerby commented:
Sorry for the delay, I’ve been unwell. Will get to this soon.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Ana Jimenez Santamaria ***@***.***>
Sent: Wednesday, December 18, 2024 4:06:28 PM
To: todogroup/ospology ***@***.***>
Cc: Alice Sowerby ***@***.***>; Mention ***@***.***>
Subject: Re: [todogroup/ospology] [OSPO Book] Update 05-chapter.md - OSPO Metrics with CHAOSS (PR #516)


@alice-sowerby<https://github.com/alice-sowerby> is this PR ready for merge? please note that early next year (probably January or mid February) , there will be a review meeting to go through all chapters at once for final review and cross-referencing

—
Reply to this email directly, view it on GitHub<https://github.com/todogroup/ospology/pull/516#issuecomment-2551719662>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEHGYSF3QLJNZ45AGBAP4FT2GGMQJAVCNFSM6AAAAABRLNXLUSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNJRG4YTSNRWGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

alice-sowerby commented:
Looks good to me (but I cannot build locally to double check the formatting).

alice-sowerby commented:
@anajsana I can do the rebase & merge if you want me to. Unclear as to whether you prefer to do it.

anajsana commented:
Rebase and merge will work. If you are unable to make the merge this way, I can do it for you 😊 @alice-sowerby
